Sources/Swift/Helper/SentryCurrentDateProvider.swift

```diff
@@ -1,16 +1,24 @@
 @_implementationOnly import _SentryPrivate
 import Foundation
 
-@objcMembers class SentryCurrentDateProvider: NSObject {
-    
+@objc
+protocol SentryCurrentDateProvider {
```

Comment:
`m`: Maybe explain why we need the protocol here. I tend to forget these things and then maybe refactor it away in a couple of weeks and then wonder why the tests now start to fail again 😅